src/gentropy/colocalisation.py

```diff
@@ -27,20 +31,52 @@ def __init__(
             credible_set_path (str): Input credible sets path.
             study_index_path (str): Input study index path.
             coloc_path (str): Output Colocalisation path.
+            colocalisation_method (ColocalisationMethod): Colocalisation method. Available methods are: ECAVIAR, COLOC.
         """
         # Extract
-        credible_set = StudyLocus.from_parquet(
-            session, credible_set_path, recursiveFileLookup=True
+        credible_set = (
+            StudyLocus.from_parquet(
+                session, credible_set_path, recursiveFileLookup=True
+            ).filter(col("finemappingMethod") == "SuSie")
+            if colocalisation_method.value == "coloc"
+            else StudyLocus.from_parquet(
+                session, credible_set_path, recursiveFileLookup=True
+            )
         )
         si = StudyIndex.from_parquet(
             session, study_index_path, recursiveFileLookup=True
         )
 
         # Transform
+        colocalisation_class = self._get_colocalisation_class(
+            colocalisation_method.value
+        )
         overlaps = credible_set.filter_credible_set(
             CredibleInterval.IS95
         ).find_overlaps(si)
-        ecaviar_results = ECaviar.colocalise(overlaps)
+        colocalisation_results = colocalisation_class.colocalise(overlaps)  # type: ignore
 
         # Load
-        ecaviar_results.df.write.mode(session.write_mode).parquet(coloc_path)
+        colocalisation_results.df.write.mode(session.write_mode).parquet(
+            f"{coloc_path}/{colocalisation_method.value}"
+        )
+
+    @classmethod
+    def _get_colocalisation_class(cls: type[ColocalisationStep], method: str) -> type:
```

Comment:
I would move this logic to `src/method/colocalisation`. You want a mapping for co-localisation methods. That's generic not only for the step you are working on.

Reply:
Can you tell me a scenario where you'd need a mapping when you are interacting directly with the classes?
I thought it made sense in the context of the step, where you want to abstract the logic from the used method.